packages/components/src/segmented-control/component.tsx

```diff
@@ -0,0 +1,122 @@
+/**
+ * External dependencies
+ */
+import { cx } from 'emotion';
+// eslint-disable-next-line no-restricted-imports
+import { RadioGroup, useRadioState } from 'reakit';
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { useRef } from '@wordpress/element';
+import { useMergeRefs, useResizeObserver } from '@wordpress/compose';
+
+/**
+ * Internal dependencies
+ */
+/**
+ * Internal dependencies
+ */
+import { contextConnect, useContextSystem } from '../ui/context';
+import { View } from '../view';
+import * as styles from './styles';
+import { useUpdateEffect } from '../utils/hooks';
+import Backdrop from './segmented-control-backdrop';
+import Button from './segmented-control-button';
+import type { SegmentedControlProps } from './types';
+
+function SegmentControl( props: any, forwardedRef: any ) {
```

Comment:
We'll definitely want to add proper types for `props` and `forwardedRef`. See other components for examples 🙂